src/dodal/beamlines/i04.py

```diff
@@ -359,10 +365,28 @@ def oav_to_redis_forwarder() -> OAVToRedisForwarder:
 @device_factory()
 def diamond_filter() -> DiamondFilter[I04Filters]:
     """Get the i04 diamond filter device, instantiate it if it hasn't already been.
-    If this is called when already instantiated in i03, it will return the existing object.
+    If this is called when already instantiated in i04, it will return the existing object.
     """
     return DiamondFilter[I04Filters](
         prefix=f"{PREFIX.beamline_prefix}-MO-FLTR-01:",
         name="diamond_filter",
         data_type=I04Filters,
     )
+
+
+@device_factory()
+def zocalo() -> ZocaloResults:
+    """Get the i04 ZocaloResults device, instantiate it if it hasn't already been.
+    If this is called when already instantiated in i04, it will return the existing object.
+    """
+    return ZocaloResults()
```

Comment:
After testing we noticed that this device has a channel which defaults to `"xrc.i03"`. We should instantiate it with 

```suggestion
    return ZocaloResults("xrc.i04")
```

Reply:
Have added this

Reply:
This needs to be a named argument, `return ZocaloResults(channel="xrc.i04")`